src/main/java/org/mule/module/apikit/deserializing/ArrayHeaderAttributeDeserializer.java

```diff
@@ -32,33 +24,38 @@ public ArrayHeaderAttributeDeserializer(ArrayHeaderDelimiter arrayHeaderDelimite
 
   @Override
   public List<String> deserializeValue(String attributeValue) {
-    if (isBlank(attributeValue)) {
-      return emptyList();
-    }
     final char delimiter = arrayHeaderDelimiter.getDelimiterValue().charAt(0);
     char[] chars = attributeValue.toCharArray();
     StringBuffer curVal = new StringBuffer();
     boolean inQuotes = false;
     int curlyBracesStackCount = 0;
     boolean startCollectChar = false;
+    boolean escapedChar = false;
 
-    List<String> headerValues = new ArrayList<>();
+    List<String> values = new ArrayList<>();
     for (char ch : chars) {
       if (inQuotes) {
         startCollectChar = true;
-        if (ch == DOUBLE_QUOTES) {
-          inQuotes = curlyBracesStackCount > 0 ? true : false;
-          if (inQuotes) {
+        if (ch == '\\') {
+          escapedChar = true;
+        } else if (ch == DOUBLE_QUOTES) {
+          if (escapedChar) {
             curVal.append(ch);
+          } else {
+            inQuotes = false;
           }
+          escapedChar = false;
         } else if (ch == OPENING_CURLY_BRACE) {
           curlyBracesStackCount++;
           curVal.append(ch);
+          escapedChar = false;
         } else if (ch == CLOSING_CURLY_BRACE) {
           curlyBracesStackCount--;
           curVal.append(ch);
+          escapedChar = false;
         } else {
           curVal.append(ch);
+          escapedChar = false;
         }
```

Comment:
I think that strings like `asd"{{{"asd}}}` will not work as expected. It also looks a bit weird that `"\{\{"` is treated exactly as `"{{"`

Also, maybe cleaner (don't know):
```suggestion
        if (ch == '\\') {
          escapedChar = true;
        } else if (ch == DOUBLE_QUOTES && !escapedChar) {
          inQuotes = false;
        } else {
          if (ch == OPENING_CURLY_BRACE) {
            curlyBracesStackCount++;
          } else if (ch == CLOSING_CURLY_BRACE) {
            curlyBracesStackCount--;
          }
          escapedChar = false;
          curVal.append(ch);
        }
```

Reply:
I updated the logic and added some tests around the cases that you mentioned